# CLEANING EVENTS

In [35]:
import pandas as pd
import numpy as np

In [36]:
events = pd.read_csv('events.csv')
events.head()

id  user_id  sequence_number                            session_id  \
0  2198523      NaN                3  83889ed2-2adc-4b9a-af5d-154f6998e778   
1  1773216      NaN                3  7a3fc3f2-e84f-44fe-8876-eff76741f7a3   
2  2380515      NaN                3  13d9b2fb-eee1-43fd-965c-267b38dd7125   
3  2250597      NaN                3  96f1d44e-9621-463c-954c-d8deb7fffe7f   
4  1834446      NaN                3  d09dce10-a7cb-47d3-a9af-44975566fa03   

                  created_at       ip_address          city      state  \
0  2021-06-17 17:30:00+00:00    138.143.9.202     São Paulo  São Paulo   
1  2020-08-07 08:41:00+00:00    85.114.141.79  Santa Isabel  São Paulo   
2  2021-02-15 18:48:00+00:00  169.250.255.132     Mairiporã  São Paulo   
3  2022-03-30 10:56:00+00:00   137.25.222.160       Cajamar  São Paulo   
4  2019-09-05 01:18:00+00:00    161.114.4.174     São Paulo  São Paulo   

  postal_code browser traffic_source      uri event_type  
0   02675-031  Chrome        Adwords  /cancel     cancel  
1   07500-000  Safari        Adwords  /cancel     cancel  
2   07600-000      IE        Adwords  /cancel     cancel  
3   07750-000  Chrome        Adwords  /cancel     cancel  
4   09581-680  Chrome          Email  /cancel     cancel

From viewing the head of the data, we can see that 'city' and 'state' contain information from places outside of the United States.

In [37]:
events.shape

(2431963, 13)

Events has 2431963 rows and 13 columns before removal of non-US countries. 

In [38]:
events['state'].unique().shape

(231,)

Events has 231 unique states before removal of non-US countries.

In [39]:
states = [
    "Alabama", "Alaska", "Arizona", "Arkansas", "California", "Colorado", 
    "Connecticut", "Delaware", "Florida", "Georgia", "Hawaii", "Idaho", 
    "Illinois", "Indiana", "Iowa", "Kansas", "Kentucky", "Louisiana", 
    "Maine", "Maryland", "Massachusetts", "Michigan", "Minnesota", 
    "Mississippi", "Missouri", "Montana", "Nebraska", "Nevada", 
    "New Hampshire", "New Jersey", "New Mexico", "New York", 
    "North Carolina", "North Dakota", "Ohio", "Oklahoma", "Oregon", 
    "Pennsylvania", "Rhode Island", "South Carolina", "South Dakota", 
    "Tennessee", "Texas", "Utah", "Vermont", "Virginia", 
    "Washington", "West Virginia", "Wisconsin", "Wyoming"
]

df_states = pd.DataFrame(states, columns=["State"])
print(df_states)

             State
0          Alabama
1           Alaska
2          Arizona
3         Arkansas
4       California
5         Colorado
6      Connecticut
7         Delaware
8          Florida
9          Georgia
10          Hawaii
11           Idaho
12        Illinois
13         Indiana
14            Iowa
15          Kansas
16        Kentucky
17       Louisiana
18           Maine
19        Maryland
20   Massachusetts
21        Michigan
22       Minnesota
23     Mississippi
24        Missouri
25         Montana
26        Nebraska
27          Nevada
28   New Hampshire
29      New Jersey
30      New Mexico
31        New York
32  North Carolina
33    North Dakota
34            Ohio
35        Oklahoma
36          Oregon
37    Pennsylvania
38    Rhode Island
39  South Carolina
40    South Dakota
41       Tennessee
42           Texas
43            Utah
44         Vermont
45        Virginia
46      Washington
47   West Virginia
48       Wisconsin
49         Wyoming


Dataframe of all US states for filtration purposes

In [40]:
US_events = events[events['state'].str.lower().isin(df_states['State'].str.lower().tolist())]

New dataframe for US events, filtered using the 'df_states' dataframe to only include rows with states in the US. 
.str.lower() was used to make sure capitalization didn't matter.

In [41]:
US_events.shape

(541963, 13)

After removal of non-US states, US_events contains 541963 rows and 13 columns.

In [44]:
split = np.array_split(US_events, 6)
for i in range(6):
    split[i].to_csv(f'US_events_{i}.csv', index=False)

/Users/paigecaskey/Desktop/work/code study/pyenvy/lib/python3.10/site-packages/numpy/_core/fromnumeric.py:57: FutureWarning: 'DataFrame.swapaxes' is deprecated and will be removed in a future version. Please use 'DataFrame.transpose' instead.
  return bound(*args, **kwds)


Export US_events to 6 different files for space purposes.